lib/ansible/plugins/connection/lxc.py

```diff
@@ -0,0 +1,225 @@
+from ansible.plugins.connection import ConnectionBase
+from ansible.utils.unicode import to_bytes
+
+HAS_LIBLXC = False
+try:
+    import lxc as _lxc
+    HAS_LIBLXC = True
+except ImportError:
+    pass
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class Connection(ConnectionBase):
+    ''' Local lxc based connections '''
+
+    transport = 'lxc'
+    has_pipelining = True
+    become_methods = frozenset(C.BECOME_METHODS)
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin, *args, **kwargs)
+
+        self.container_name = self._play_context.remote_addr
+        self.container = None
+
+    def _connect(self):
+        ''' connect to the lxc; nothing to do here '''
+        super(Connection, self)._connect()
+
+        if not HAS_LIBLXC:
+            msg = "lxc bindings for python2 are not installed"
+            raise errors.AnsibleError(msg)
+
+        if self.container:
+            return
+
+        display.vvv("THIS IS A LOCAL LXC DIR", host=self.container_name)
+        self.container = _lxc.Container(self.container_name)
+        if self.container.state == "STOPPED":
+            raise errors.AnsibleError("%s is not running" % self.container_name)
+
+    def _communicate(self, pid, in_data, stdin, stdout, stderr):
+        buf = { stdout: [], stderr: [] }
+        read_fds = [stdout, stderr]
+        if in_data:
+            write_fds = [stdin]
+        else:
+            write_fds = []
+        while len(read_fds) > 0 or len(write_fds) > 0:
+            try:
+                ready_reads, ready_writes, _ = select.select(read_fds, write_fds, [])
+            except select.error as e:
+                if e.args[0] == errno.EINTR:
+                    continue
+                raise
+            for fd in ready_writes:
+                in_data = in_data[os.write(fd, in_data):]
+                if len(in_data) == 0:
+                    write_fds.remove(fd)
+            for fd in ready_reads:
+                data = os.read(fd, 32768)
+                if not data:
+                    read_fds.remove(fd)
+                buf[fd].append(data)
+
+        (pid, returncode) = os.waitpid(pid, 0)
+
+        return returncode, b"".join(buf[stdout]), b"".join(buf[stderr])
+
+    def _set_nonblocking(self, fd):
+        flags = fcntl.fcntl(fd, fcntl.F_GETFL) | os.O_NONBLOCK
+        fcntl.fcntl(fd, fcntl.F_SETFL, flags)
+        return fd
+
+    def exec_command(self, cmd, in_data=None, sudoable=False):
+        ''' run a command on the chroot '''
+        super(Connection, self).exec_command(cmd, in_data=in_data, sudoable=sudoable)
+
+        if C.DEFAULT_EXECUTABLE:
+            executable = C.DEFAULT_EXECUTABLE.split()[0]
+        else:
+            executable = '/bin/sh'
+        local_cmd = [to_bytes(executable, errors='strict'), '-c', to_bytes(cmd, errors='strict')]
```

Comment:
New connection plugins should be using the recently added ```self._play_context.executable``` for executable. Take a look at the lxd plugin for an example.